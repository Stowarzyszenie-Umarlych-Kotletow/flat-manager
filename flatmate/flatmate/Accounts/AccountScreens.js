import {Keyboard, Text, View, TextInput, TouchableWithoutFeedback, Alert, KeyboardAvoidingView} from 'react-native';
import {Button} from 'react-native-elements';
import * as React from "react";
import styles from "./styles";

export function LoginScreen() {
    return (
        <KeyboardAvoidingView style={styles.containerView} behavior="padding">

            {/*<TouchableWithoutFeedback onPress={Keyboard.dismiss}>*/}
            <View style={styles.accScreenContainer}>
                <View style={styles.accFormView}>
                    <Text style={styles.logoText}>Login to Flatmate</Text>
                    <TextInput placeholder="Username" placeholderColor="#c4c3cb" style={styles.accFormTextInput}/>
                    <TextInput placeholder="Password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                               secureTextEntry={true}/>
                    <Button
                        buttonStyle={styles.accButton}
                        // onPress={() => this.onLoginPress()}
                        title="Login"
                    />
                </View>
            </View>
            {/*</TouchableWithoutFeedback>*/}
        </KeyboardAvoidingView>
    );
}

export function RegisterScreen() {
    return (
        <KeyboardAvoidingView style={styles.containerView} behavior="padding">
            {/*<TouchableWithoutFeedback onPress={Keyboard.dismiss}>*/}
            <View style={styles.accScreenContainer}>
                <View style={styles.accFormView}>
                    <Text style={styles.logoText}>Register to Flatmate</Text>
                    <TextInput placeholder="Username" placeholderColor="#c4c3cb" style={styles.accFormTextInput}/>
                    <TextInput placeholder="Password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                               secureTextEntry={true}/>
                    <TextInput placeholder="Retype password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                               secureTextEntry={true}/>
                    <Button
                        buttonStyle={styles.accButton}
                        // onPress={() => this.onLoginPress()}
                        title="Register"
                    />
                </View>
            </View>
            {/*</TouchableWithoutFeedback>*/}
        </KeyboardAvoidingView>
    );
}


export function HomeScreen({navigation}) {
    return (
        <View style={styles.accScreenContainer}>
            <View style={styles.accFormView}>
                <Text style={styles.logoText}>Flatmate</Text>
                <Button
                    buttonStyle={styles.accButton}
                    title="Login"
                    onPress={() => navigation.navigate('LoginScreen')}
                />
                <Button
                    buttonStyle={styles.accButton}
                    title="Register"
                    onPress={() => navigation.navigate('RegisterScreen')}
                />
            </View>
        </View>
    );
}