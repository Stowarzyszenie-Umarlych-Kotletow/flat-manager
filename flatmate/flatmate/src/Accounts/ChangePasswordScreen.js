import {Text, TextInput, View} from "react-native";
import styles from "../native_elements_styles";
import {Button} from "react-native-elements";
import * as React from "react";
import '../bootstrap_style.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.js';

export function ChangePasswordScreen({navigation}) {
    function onChangePasswordPress() {
        console.log('saving data')
        navigation.pop()
    }

    return (
        <View style={styles.accScreenContainer}>
            <View style={styles.accFormView}>
                <Text style={styles.logoText}>Change password</Text>

                <TextInput placeholder="Old password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                           secureTextEntry={true}/>

                <TextInput placeholder="New password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                           secureTextEntry={true}/>

                <TextInput placeholder="Retype password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                           secureTextEntry={true}/>

                <Button
                    buttonStyle={styles.bluButton}
                    onPress={() => onChangePasswordPress()}
                    title="Save"
                />
            </View>
        </View>
    );
}