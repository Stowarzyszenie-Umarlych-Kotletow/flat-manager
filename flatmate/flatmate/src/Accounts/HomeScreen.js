import {Text, View} from 'react-native';
import {Button} from 'react-native-elements';
import * as React from "react";
import styles from "../static/native_elements_styles";

export function HomeScreen({
                               navigation
                           }) {
    return (<View style={styles.accScreenContainer}>
            <View style={styles.accFormView}>
                <Text style={styles.logoText}>Flatmate</Text>

                <Button
                    buttonStyle={styles.bluButton}
                    title="Login"
                    onPress={() => navigation.navigate('LoginScreen')}
                />

                <Button
                    buttonStyle={styles.bluButton}
                    title="Register"
                    onPress={() => navigation.navigate('RegisterScreen')}
                />
            </View>
        </View>);
}