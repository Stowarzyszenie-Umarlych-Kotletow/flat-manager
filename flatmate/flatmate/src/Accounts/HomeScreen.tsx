import {Text, View} from 'react-native';
import {Button} from 'react-native-elements';
import * as React from "react";
import styles from "../static/styles";

export function HomeScreen({navigation}) {
    return (<View style={styles.accScreenContainer}>
        <Text style={styles.loginLogoText}>Flatmate</Text>

        <Button
            buttonStyle={styles.blueButton}
            title="Login"
            onPress={() => navigation.navigate('LoginScreen')}
        />

        <Button
            buttonStyle={styles.blueButton}
            title="Register"
            onPress={() => navigation.navigate('RegisterScreen')}
        />
    </View>);
}