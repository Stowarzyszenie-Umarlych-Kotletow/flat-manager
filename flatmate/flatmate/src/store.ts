import { combineReducers, configureStore, applyMiddleware } from '@reduxjs/toolkit'
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import storage from '@react-native-async-storage/async-storage';
import auth from "./features/auth"
import flat from './features/flat';
import { api } from './features/api/api';
import { setupListeners } from '@reduxjs/toolkit/dist/query';

const rootReducer = combineReducers({
  auth: auth.reducer,
  flat: flat.reducer,
  [api.reducerPath]: api.reducer
});
const persistConfig = {
  key: 'root',
  storage
};
const persistedReducer = persistReducer(persistConfig, rootReducer);


const store = configureStore({
  reducer: persistedReducer,
  devTools: true,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }).concat(api.middleware)
});

setupListeners(store.dispatch);

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>() // Export a hook that can be reused to resolve types
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export default store;