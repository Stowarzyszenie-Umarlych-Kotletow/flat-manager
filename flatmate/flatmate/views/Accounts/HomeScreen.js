import {Text, View} from 'react-native';
import {Button} from 'react-native-elements';
import * as React from "react";
import styles from "./styles";

export function HomeScreen({
                               navigation
                           }) {
    return (<View style={styles.accScreenContainer}>
            <View style={styles.accFormView}>
                <Text style={styles.logoText}>Flatmate</Text>
                <Button
                    buttonStyle={styles.accButton}
                    title="Login"
                    onPress={() => navigation.navigate('LoginScreen')}
                />
                <Button
                    buttonStyle={styles.accButton}
                    title="Register"
                    onPress={() => navigation.navigate('RegisterScreen')}
                />
            </View>
        </View>);
}