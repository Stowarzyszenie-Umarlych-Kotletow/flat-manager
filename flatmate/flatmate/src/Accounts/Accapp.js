import * as React from "react";
import styles from "../native_elements_styles";
import {Text, View} from "react-native";
import logo from '../logo.svg'
import {Button} from "react-native-elements";
import {createStackNavigator} from "@react-navigation/stack";
import {HomeScreen} from "./HomeScreen";
import {RegisterScreen} from "./RegisterScreen";
import {LoginScreen} from "./LoginScreen";
import {ManageScreen} from "./ManageScreen";
import {ChangePasswordScreen} from "./ChangePasswordScreen";
const AccStack = createStackNavigator();

export function AccApp({navigation}) {
    return (
        <AccStack.Navigator
            initialRouteName="HomeScreen"
            screenOptions={{
                headerMode: 'screen',
                headerTintColor: 'white',
                headerStyle: {backgroundColor: 'black'},
                // headerLeft: () =>(<MyIcon/>)
                animationEnabled: true,

            }}>
            <AccStack.Screen name="HomeScreen" component={HomeScreen} options={{
                title: 'Flatmate'
            }}/>
            <AccStack.Screen name="RegisterScreen" component={RegisterScreen} options={{
                title: 'Register'
            }}/>
            <AccStack.Screen name="LoginScreen" component={LoginScreen} options={{
                title: 'Login'
            }}/>
            <AccStack.Screen name="ManageScreen" component={ManageScreen} options={{
                title: 'Manage account'
            }}/>
            <AccStack.Screen name="ChangePasswordScreen" component={ChangePasswordScreen} options={{
                title: 'Change Password'
            }}/>
        </AccStack.Navigator>
    )
}