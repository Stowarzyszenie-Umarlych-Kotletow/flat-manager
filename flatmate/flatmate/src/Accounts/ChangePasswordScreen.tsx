import {Text, TextInput, View} from "react-native";
import styles from "../static/styles";
import {Button} from "react-native-elements";
import {Controller, useForm} from "react-hook-form";
import * as React from "react";
import {useState} from "react";
import {useAppDispatch} from "../store";
import {ChangePasswordRequest} from "../models/api/account";


export function parseData(data) {
    const parsedData: ChangePasswordRequest = {
        "password": data.newPassword,
    }
    return parsedData;
}

export function ChangePasswordScreen() {
    const {control, handleSubmit, formState: {errors}} = useForm({
        defaultValues: {
            currentPassword: '',
            newPassword: '',
            newPasswordConfirm: '',
        },
    });
    const [showIncorrectPasswordWarning, setShowIncorrectPasswordWarning] = useState(false);
    const [showDifferentPasswordsWarning, setShowDifferentPasswordsWarning] = useState(false);
    
    const dispatch = useAppDispatch();

    async function onChangePasswordPress(data) {

        if (data.newPassword != data.newPasswordConfirm || !data.newPassword) {
            setShowDifferentPasswordsWarning(true);
        } else {
            setShowDifferentPasswordsWarning(false);
        }
        let parsedData = parseData(data);
        // backend connection
        // change password
        setShowIncorrectPasswordWarning(false);
        // todo
    }

    return (
        <View style={styles.accScreenContainer}>
            <Text style={styles.logoText}>Change password</Text>

            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({field: {onChange, onBlur, value}}) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Current password"
                        secureTextEntry={true}
                    />
                )}
                name="currentPassword"
            />

            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({field: {onChange, onBlur, value}}) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="New password"
                        secureTextEntry={true}
                    />
                )}
                name="newPassword"
            />

            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({field: {onChange, onBlur, value}}) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Confirm new password"
                        secureTextEntry={true}
                    />
                )}
                name="newPasswordConfirm"
            />

            {!showIncorrectPasswordWarning ? null :
                <Text style={styles.warningText}> Incorrect Current Password</Text>}
            {!showDifferentPasswordsWarning ? null :
                <Text style={styles.warningText}> Passwords do not match <br/> or <br/>You are trying to push empty
                    password </Text>}
            <Button
                buttonStyle={styles.greenButton}
                title="Change Password"
                onPress={handleSubmit(onChangePasswordPress)}
            />
        </View>
    );
}