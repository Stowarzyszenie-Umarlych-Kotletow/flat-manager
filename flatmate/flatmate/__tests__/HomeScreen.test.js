import React from 'react';
import renderer from 'react-test-renderer';
import {HomeScreen} from "../src/Accounts/HomeScreen";
import store from "../src/store";
import {Provider} from 'react-redux';

test('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><HomeScreen/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});