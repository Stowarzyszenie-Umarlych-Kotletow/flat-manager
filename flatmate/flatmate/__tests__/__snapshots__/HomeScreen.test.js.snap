// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "800",
          "marginBottom": 30,
          "marginTop": 150,
          "textAlign": "center",
        }
      }
    >
      Flatmate
    </Text>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3897f1",
              "borderColor": "#2089dc",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 10,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3897f1",
              "borderColor": "#2089dc",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 10,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Register
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
